book/P2C9_calib.qmd

```diff
@@ -143,3 +143,15 @@ For the models in @fig-eval-dcalib, it is clear that the relative risk tree is n
 Whilst the D-calibration for the Cox model is smaller than that of the random forest, the difference is unlikely to be significant, as is seen in the overlapping curves in the figure.
 
 The next chapter will look at scoring rules, which provides a more concrete method to analytically compare the predicted distributions from survival models.
+
+## Beyond right-censoring
+
+Using graphical plots is simplest for measuring calibration when left-censoring, interval-censoring, and/or truncation are involved.
+In these contexts, predicted survival functions can be compared to the 'true' survival probability estimated with the Kaplan-Meier (@sec-calib-km) by using definitions of the Kaplan-Meier estimator that incorporate other censoring and truncation types as required.
+
+For left- and interval-censoring, the Kaplan-Meier estimator is substituted with the nonparametric maximum likelihood estimator (NPMLE) [@Zhigang2010], which is the method to estimate the likelihood function under the assumption of left- or interval-censoring data as in @sec-surv-estimation-param.
+The most common algorithm to do so is the Turnbull algorithm [@Turnbull1974], which is not discussed in detail here but is widely available in software implementations.
+
+In the presence of left-truncation, the Kaplan-Meier is calculated using the left-truncated risk-set definition [@McGough2021], which removes observations from being 'at risk' if their left-truncation time has yet to occur (@eq-risk-lt).
```

Comment:
I think this should probalby go into the survival chapter, if it's not there already

Reply:
Yeah let's discuss on Thurs